sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -338,22 +490,39 @@ private final class GpuSemaphore() extends Logging {
   import GpuSemaphore._
 
   type GpuBackingSemaphore = PrioritySemaphore[Long]
-  private val semaphore = new GpuBackingSemaphore(MAX_PERMITS, GpuSemaphore.DEFAULT_PRIORITY)
+  private val semaphore = new GpuBackingSemaphore(computeMaxPermits(),
+    GpuSemaphore.DEFAULT_PRIORITY)
   // A map of taskAttemptId => semaphoreTaskInfo.
   // This map keeps track of all tasks that are both active on the GPU and blocked waiting
   // on the GPU.
   private val tasks = new ConcurrentHashMap[Long, SemaphoreTaskInfo]
+  private val stageEstimators = {
+    val lru = new util.LinkedHashMap[Int, GpuStageMemoryEstimator]() {
+      override def removeEldestEntry(entry: Map.Entry[Int, GpuStageMemoryEstimator]): Boolean = {
+        //System.err.println(s"SHOULD REMOVE??? $size > 100? FOR STAGE ${entry.getKey}")
+        size > 100
```

Comment:
Is this because we don't know when the stage is finished (and thus we don't know when to clean up the entry for the finished stage)?

Reply:
yes